### PR TITLE
revert: default LSP mappings

### PR DESCRIPTION
**Pull request by gpanders**
Revert the default LSP mappings before the 0.10 release as these might need some further consideration. In particular, it's not clear if "c" prefixed maps in Normal mode are acceptable as defaults since they interfere with text objects or operator ranges.

We will re-introduce default mappings at the beginning of the 0.11 release cycle, this reversion is only for the imminent 0.10 release.

Ref: https://github.com/neovim/neovim/issues/28634

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

**Comment by zeertzjq** [comment]
Also: Insert mode Ctrl-S mapping conflicts with vim-surround.

File: runtime/doc/news.txt [hunk 1]
```diff
@@ -418,12 +418,6 @@ The following changes to existing APIs or features add new behavior.
   • 'shortmess' includes the "C" flag.
   • 'grepprg' uses the -H and -I flags for grep by default,
     and defaults to using ripgrep if available.
-  • |crn| in Normal mode maps to |vim.lsp.buf.rename()|.
-  • |crr| in Normal mode maps to |vim.lsp.buf.code_action()|.
-  • |v_CTRL-R_CTRL-R| in Visual mode maps to |vim.lsp.buf.code_action()|.
-  • "gr" in Normal mode maps to |vim.lsp.buf.references()| |gr-default|
-  • |i_CTRL-S| in Insert mode maps to |vim.lsp.buf.signature_help()|
-  • "]d" and "[d" in Normal mode map to |vim.diagnostic.goto_next()| and
```
Comment (lewis6991) [inline runtime/doc/news.txt:426]: Was this line supposed to be removed?

Comment (gpanders) [reply runtime/doc/news.txt:426]: No :(

**Comment by wookayin** [comment]
Ref #28500

Is the helptag *gr-default* also supposed to be removed (doc/change.txt)?

**Comment by gpanders** [comment]
@wookayin Yes, I thought I had done that but apparently I did not (looks like I included it in https://github.com/neovim/neovim/pull/28650 by mistake instead).

**Pull request merged**